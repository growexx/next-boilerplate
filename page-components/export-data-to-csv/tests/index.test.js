import { fireEvent, render, screen } from '@testing-library/react'

import ExportDataToCsv from '../index'

describe('<ExportDataToCsv />', () => {
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild }
    } = render(<ExportDataToCsv />)
    expect(firstChild).toMatchSnapshot()
  })
  it('should trigger export', () => {
    render(<ExportDataToCsv />)
    const checkbox = screen.getByRole('checkbox', { name: /select all/i })
    fireEvent.click(checkbox)
    render(<ExportDataToCsv />)
    const buttonElement = screen.getAllByTestId('ExportButton')
    fireEvent.click(buttonElement[0])
    expect(buttonElement[0].tagName).toEqual('BUTTON')
  })
})
