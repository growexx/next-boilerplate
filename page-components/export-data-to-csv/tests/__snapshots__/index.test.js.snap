// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExportDataToCsv /> Should render and match the snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB kTuyhl"
  >
    <div
      class="sc-gueYoa hAnktJ"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-ph9edi ant-btn-primary"
        data-testid="ExportButton"
        disabled=""
        type="button"
      >
        <span>
          Export Data
        </span>
      </button>
    </div>
    <div
      class="ant-table-wrapper css-dev-only-do-not-override-ph9edi"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-ph9edi"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-scroll-horizontal"
            data-testid="DataTable"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
                style="overflow-x: auto; overflow-y: hidden;"
              >
                <table
                  style="width: 350px; min-width: 100%; table-layout: auto;"
                >
                  <colgroup>
                    <col
                      class="ant-table-selection-col"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell ant-table-selection-column"
                        scope="col"
                      >
                        <div
                          class="ant-table-selection"
                        >
                          <label
                            class="ant-checkbox-wrapper css-dev-only-do-not-override-ph9edi"
                          >
                            <span
                              class="ant-checkbox css-dev-only-do-not-override-ph9edi"
                            >
                              <input
                                aria-label="Select all"
                                class="ant-checkbox-input"
                                type="checkbox"
                                value=""
                              />
                              <span
                                class="ant-checkbox-inner"
                              />
                            </span>
                          </label>
                        </div>
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Name
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Age
                      </th>
                      <th
                        class="ant-table-cell"
                        scope="col"
                      >
                        Address
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      aria-hidden="true"
                      class="ant-table-measure-row"
                      style="height: 0px; font-size: 0px;"
                    >
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                      <td
                        style="padding: 0px; border: 0px; height: 0px;"
                      >
                        <div
                          style="height: 0px; overflow: hidden;"
                        >
                           
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="1"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper css-dev-only-do-not-override-ph9edi"
                        >
                          <span
                            class="ant-checkbox css-dev-only-do-not-override-ph9edi"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        John Brown
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        New York No. 1 Lake Park
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="2"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper css-dev-only-do-not-override-ph9edi"
                        >
                          <span
                            class="ant-checkbox css-dev-only-do-not-override-ph9edi"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Jim Green
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        42
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        London No. 1 Lake Park
                      </td>
                    </tr>
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key="3"
                    >
                      <td
                        class="ant-table-cell ant-table-selection-column"
                      >
                        <label
                          class="ant-checkbox-wrapper css-dev-only-do-not-override-ph9edi"
                        >
                          <span
                            class="ant-checkbox css-dev-only-do-not-override-ph9edi"
                          >
                            <input
                              class="ant-checkbox-input"
                              type="checkbox"
                              value=""
                            />
                            <span
                              class="ant-checkbox-inner"
                            />
                          </span>
                        </label>
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Joe Black
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        32
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        Sidney No. 1 Lake Park
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
